examples/t_factory.py

```diff
@@ -11,31 +12,28 @@
 from guppylang.std.quantum_functional import (
     cz,
     h,
-    rx,
+    ry,
     rz,
 )
 
 phi = np.arccos(1 / 3)
 
 
-@guppy
-def ry(q: qubit @owned, theta: angle) -> qubit:
-    q = rx(q, pi / 2)
-    q = rz(q, theta + pi)
-    q = rx(q, pi / 2)
-    return rz(q, pi)
-
 # Preparation of approximate T state, from https://arxiv.org/abs/2310.12106
 @guppy
-def prepare_approx(q: qubit @owned) -> qubit:
+def prepare_approx(q: qubit @ owned) -> qubit:
     q = ry(q, angle(py(phi)))
     return rz(q, pi / 4)
 
 
 # The inverse of the [[5,3,1]] encoder in figure 3 of https://arxiv.org/abs/2208.01863
 @guppy
 def distill(
-    target: qubit @owned, q0: qubit @owned, q1: qubit @owned, q2: qubit @owned, q3: qubit @owned
+    target: qubit @ owned,
+    q0: qubit @ owned,
+    q1: qubit @ owned,
```

Comment:
An alertnative would be to take an array of qubits as input here and index into it for the cz gates.

Reply:
added in [c6e96b5](https://github.com/CQCL/guppylang/pull/817/commits/c6e96b546355f6481329bba6509b42883691eaed). I can revert if the old way is preferred.